src/backend/distributed/commands/citus_add_local_table_to_metadata.c

```diff
@@ -1019,6 +1034,41 @@ GetDropTriggerCommand(Oid relationId, char *triggerName)
 }
 
 
+/*
+ * DropViewsOnTable drops the views that depend on the given relation.
+ */
+static void
+DropViewsOnTable(Oid relationId)
+{
+	List *views = GetDependingViews(relationId);
+	List *reverseOrderedViews = NIL;
+
+	/*
+	 * GetDependingViews returns views in the dependency order. We should drop views
+	 * in the reversed order since dropping views can cascade to other views below.
+	 */
+	Oid viewId = InvalidOid;
```

Comment:
Minor: can we have a helper `ReverseOidList` function similar to `ActiveSubXactContexts ` but for oids? `list_insert_nth_oid` looks a little harder to follow